engine/server/session.go

```diff
@@ -1252,6 +1252,13 @@ func (srv *Server) ServeModule(ctx context.Context, mod *core.Module) error {
 	client.stateMu.Lock()
 	defer client.stateMu.Unlock()
 
+	// don't add the same module twice
+	// This can happen with generated clients since all remote dependencies are added
+	// on each connection and this could happen multiple times.
+	if client.deps.LookupDep(mod.Name()) {
+		return nil
+	}
```

Comment:
Ahh I see. That's kind of reasonable I guess :thinking: 

But, as mentioned above, this logic is probably a bit more complex, since it should handle trying to serve two different versions of the same module (this should error out). Or if there's a module with this name already, and *that* module doesn't have the same module source.

Reply:
So what should I do? I'm not sure I understand when I should error or not.

Reply:
No, this check won't error on the last case right? It would just ignore this, and continue without serving `module-y`.

We actually need to check that the looked-up dependency is the same as the `mod` that's being passed in.